src/meta/src/stream/scale.rs

```diff
@@ -835,17 +886,42 @@ where
                 .await;
         }));
 
+        self.source_manager.pause_tick().await;
+
+        let source_manager_ref = self.source_manager.clone();
+
+        // Note: If no error occurs, we need to manually resume the tick
+        revert_funcs.push(Box::pin(async move {
+            tracing::warn!("source manager tick resumed due to error");
+            source_manager_ref.resume_tick().await;
+        }));
+
         tracing::trace!("reschedule plan: {:#?}", reschedule_fragment);
 
-        self.barrier_scheduler
+        let result = self
+            .barrier_scheduler
             .run_multiple_commands(vec![
                 Command::Plain(Some(Mutation::Pause(PauseMutation {}))),
                 Command::RescheduleFragment(reschedule_fragment),
                 Command::Plain(Some(Mutation::Resume(ResumeMutation {}))),
             ])
-            .await?;
+            .await;
 
-        Ok(())
+        if result.is_ok() {
+            if !stream_source_actor_splits.is_empty() {
+                self.source_manager
+                    .patch_update(
+                        None,
+                        Some(stream_source_actor_splits),
+                        Some(stream_source_dropped_actors),
+                    )
+                    .await?;
```

Comment:
Should we do this in `Command::post_collect` or here? I'm not sure whether there could be some race cases.

Reply:
+1, `patch_update` should be called in `post_collect` and those meta data should be updated in a same transaction as long as table fragments I think. If fails or meta crash before this, there will be some inconsistency. Add that's also needed for `create_materialized_view` in stream manager.

Reply:
Let's add a param `txn: &mut Transaction` for `patch_update`, so that it can be called by both ticker inside source manager and barrier manager.